lib/vault.js

```diff
@@ -0,0 +1,71 @@
+const { exec } = require('child_process');
+const fs = require('fs');
+
+const service = process.env.SERVICE_SYSTEM_CODE;
+const region = (process.env.REGION || 'LOCAL').toLowerCase();
+const vaultPath = `secret/teams/origami/${service}/${region}`;
+const envFile = '.env';
+
+if (!service) {
+    console.warn(
+        'Could not load enviroment variables from Vault. Service name is undefined.',
+        'Set the enviroment variable "SERVICE_SYSTEM_CODE" in your Makefile.'
+    );
+    process.exit(1);
+}
+
+// Update an existing .env file with enviroment variables from Vault.
+exec(`vault read -format=json ${vaultPath}`, (err, stdout, stderr) => {
+    const result = err ? null : JSON.parse(stdout);
+
+    // Handle errors.
+    if (err || !result.data) {
+        console.warn(
+            `Could not load enviroment variables from Vault (${vaultPath}).`,
+            (stdout ? `\nstdout: ${stdout}` : ''),
+            (stderr ? `\nstderr: ${stderr}` : '')
+        );
+        process.exit(1);
+    }
+
+    // Get secrets object.
+    const existingComments = {};
+    const existingSecrets = fs.readFileSync(envFile).toString().trim().split('\n').reduce((secretsObject, line, index) => {
+        if (line.trim().indexOf('#') == 0) {
+            // Line is a comment.
+            existingComments[index] = line;
+        } else {
+            // Line is a secret (enviroment variable).
+            const secretArr = (line.includes('=') ? line.split('=') : null);
+            if (secretArr) {
+                secretsObject[secretArr[0]] = secretArr[1];
+            }
+        }
+        return secretsObject;
+    }, {});
+    const vaultSecrets = result.data;
+    const secrets = Object.assign(existingSecrets, vaultSecrets)
+
+    // Create string of secrets for .env.
+    const keys = Object.keys(secrets);
+    let envContent = '';
+    for (index = 0; index <= keys.length - 1; index++) {
```

Comment:
`index` is currently a global variable, update this to `let index = 0` and it will no longer be global scoped but block scoped